File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.stream.*;
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        Queue<int[]> q = new LinkedList<>();
        
        for(int i = 0; i< priorities.length;i++){
            int[] ad = new int[]{i, priorities[i]};
            q.add(ad);
        }
        
        Arrays.sort(priorities);
        int count = 1;
        
        for(int i = priorities.length - 1; i >= 0; i--){
            int[] qInt = q.poll();
            
            while(qInt[1] < priorities[i]){
                q.add(qInt);
                qInt = q.poll();
            }
            if(qInt[0] == location){
                break;
            }
            
            count++;
        }

        return count;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const cS = [...priorities]
    cS.sort((a,b) => a - b);
    priorities = priorities.map((v,i) => [v,i]);
    
    let cnt = 0;
    
    while(cS.at(-1) != priorities[0][0] || location != priorities[0][1]){
        if (cS.at(-1) === priorities[0][0]){
            priorities.shift()
            cS.pop()
            cnt += 1;
            continue;
        }
        priorities.push(priorities.shift());
    }
    
    return cnt + 1
}